src/lib.rs

```diff
@@ -292,50 +292,50 @@ impl<T> Storage<T> {
         }
     }
 
-    /// Iterate all components in this storage.
+    /// Iterate all components in this storage that are still referenced from outside.
+    /// ### Attention
+    /// Information about live components is updated not for all changes, but
+    /// only when you explicitly call [`sync_pending`](struct.Storage.html#method.sync_pending).
+    /// It means, you can get wrong results when calling this function before updating pending.
     #[inline]
     pub fn iter(&self) -> ReadIter<T> {
         ReadIter {
             storage: &self.inner,
-            skip_lost: false,
+            skip_lost: true,
             index: 0,
         }
     }
 
-    /// Iterate all components that are still referenced by something.
-    /// ### Attention
-    /// Information about live components is updated not for all changes, but
-    /// only when you explicitly call [`sync_pending`](struct.Storage.html#method.sync_pending).
-    /// It means, you can get wrong results when calling this function before updating pending.
+    /// Iterate all components that are stored, even if not referenced.
+    /// This can be faster than the regular `iter` for the lack of `meta` checks.
```

Comment:
I don't think user should even know about implementation details such as `meta` (there is no mention of it in documentation afaik)